[styles] Fix ditch rendering and strings

RedAuburn opened the pull request:
old on left, new right
![bitmap](https://github.com/organicmaps/organicmaps/assets/26939824/8fea455e-8214-46ba-be0a-01dfd24d6828)

pastk commented on data/strings/types_strings.txt:
```diff
@@ -4923,7 +4923,7 @@
 
-  [type.barrier.ditch]
+  [type.waterway.ditch]
     en = Moat
```
A moat is usually quite wide, deep and used as a defense structure, right?
While waterway-ditch is usually a small trench for draining.

So maybe something like a "Drainage Ditch" would be better?

Overall looking at e.g. slavic translations I can't really say which one is with water and which one is dry :)
I bet there is a similar issue for many other languages (and it probably explains why this translation inconsistency was unnoticed for a long time).

So probably it would be best first to devise better EN translations and then ask the translations team to update other languages.

RedAuburn replied:
Ah true, the correct type for moat would be `type.natural.water.moat`, but we don't include it & there are less than 2000 uses, so i'll leave the translation for the future.

RedAuburn commented on data/styles/clear/include/Basemap.mapcss:
```diff
@@ -291,6 +291,8 @@ line|z17-[waterway=ditch],
 line|z17-[barrier=ditch],
```
oops fixed

RedAuburn commented:
@organicmaps/Translations

can you provide translations for "Drainage Ditch"
many thanks :)

matheusgomesms commented:
> @organicmaps/translations
> 
> can you provide translations for "Drainage Ditch" many thanks :)

PT=PT-BR -> Vala de drenagem

jimcarst commented on data/strings/types_strings.txt:
```diff
@@ -4923,7 +4923,11 @@
+  [type.waterway.ditch]
+    ref = type.barrier.ditch
+    en = Drainage Ditch
+
```
```suggestion
nl = Sloot
```

jimcarst commented on data/strings/types_strings.txt:
```diff
@@ -4923,7 +4923,11 @@
+    en = Drainage Ditch
+
+  [type.natural.water.moat]
     en = Moat
```
```suggestion
    en = Moat
    nl = Gracht
```

jimcarst commented on data/strings/types_strings.txt:
```diff
@@ -27681,7 +27685,7 @@
-  [type.waterway.ditch]
+  [type.barrier.ditch]
     en = Ditch
     af = Sloot
```
```suggestion
    af = Sloot
    nl = Sloot
```

dvdmrtnz commented:
In Spanish it could be "Zanja de drenaje" or simply "Zanja"

RedAuburn commented:
added auto generated translations (and included already commented translations)

vng commented on data/strings/types_strings.txt:
```diff
@@ -4923,7 +4923,49 @@
+  [type.waterway.ditch]
+    ref = type.barrier.ditch
+    en = Drainage Ditch
+	ar = حفرة الصرف الصحي
```
- ref is not needed?
- align tabs

vng commented:
Should we also add natural-water-ditch type?
https://wiki.openstreetmap.org/wiki/Tag%3Awater%3Dditch

zubialevich commented:
> @organicmaps/translations
> 
> can you provide translations for "Drainage Ditch" many thanks :)

be (belarusian) => Дрэнажная канава

RedAuburn commented:
> Should we also add natural-water-ditch type? https://wiki.openstreetmap.org/wiki/Tag%3Awater%3Dditch

i suppose there's no reason not to :) also added natural.water.moat for the moat translation strings.

pastk commented on data/mapcss-mapping.csv:
```diff
@@ -513,8 +513,8 @@ healthcare|speech_therapist;438;
-deprecated|deprecated;442;x
-deprecated|deprecated;443;x
+natural|water|ditch;[natural=water][water=ditch];;name;int_name;442;
+natural|water|moat;[natural=water][water=moat];;name;int_name;443;x
```
remove the trailing `x`

pastk commented on data/styles/clear/include/priorities_2_BG-top.prio.txt:
```diff
@@ -20,8 +20,10 @@ landuse-reservoir                                   # area z12- (also has captio
+natural-water-ditch                                 # area z1- (also has caption z10-)
 natural-water-lake                                  # area z1- (also has caption z10-)
 natural-water-lock                                  # area z1- (also has caption z10-)
+natural-water-moat                                  # area z1- (also has caption z10-)
```
These are relatively small water features, please make their captions appear on z17-.

Otherwise there might be captions visible on e.g. z11, but no area water because it was filtered out due to a small polygon size.

RedAuburn replied:
fixed

pastk commented:
```
barrier.ditch
ru = Ров

water.moat
ru = Оборонительный ров
```

d4f5409d commented:
Man made ditch 🇭🇺hu = (Mesterséges) csatorna
Natural ditch 🇭🇺hu = Árok

d4f5409d commented on android/res/values-hu/strings.xml:
```diff
@@ -846,7 +846,8 @@
 	<string name="type.barrier.city_wall">Városi fal</string>
-	<string name="type.barrier.ditch">Várárok</string>
+	<string name="type.waterway.ditch">Vízelvezető árok</string>
+	<string name="type.natural.water.moat">Várárok</string>
```
Only "Árok"

RedAuburn replied:
for ditch or moat?

d4f5409d reviewed (changes requested):
https://github.com/organicmaps/organicmaps/pull/5733#discussion_r1297397191

d4f5409d commented:
We could add man made ditch too (if it does present)

Man made ditch = Mesterséges csatorna

pastk commented:
> We could add man made ditch too (if it does present)
> 
> Man made ditch = Mesterséges csatorna 🇭🇺 (hu)

There is no such a tag. Did you mean a `barrier=ditch` maybe? Its in this PR.

d4f5409d commented:
> > We could add man made ditch too (if it does present)
> > 
> > Man made ditch = Mesterséges csatorna 🇭🇺 (hu)
> 
> There is no such a tag. Did you mean a `barrier=ditch` maybe? Its in this PR.

Nah

pastk commented:
Maybe add a [`water=drain`](https://wiki.openstreetmap.org/wiki/Tag:water%3Ddrain) also? :)

biodranik commented:
Please also check translations for the ditch tunnel :) https://github.com/organicmaps/organicmaps/pull/5691/commits/141a14907a96da11dd12fb25d0d11842e10dd56e#diff-dc83cee833f96d4ab35de9bc5adb183c39800fe95659d682096c694696dc9aeaR27729

RedAuburn commented:
> Please also check translations for the ditch tunnel :)

added :)

> Maybe add a [`water=drain`](https://wiki.openstreetmap.org/wiki/Tag:water%3Ddrain) also? :)

it's already included in OM 👍

fixed review stuff

RedAuburn commented:
oh wait you're right, OM includes waterway=drain, but not water=drain. Will fix.

pastk commented on data/strings/types_strings.txt:
```diff
@@ -27719,12 +27760,12 @@
+  [type.natural.water.ditch]
     ref = type.waterway.ditch
     en = Culvert
     de = Durchlass
```
please remove all translations here (let the `ref` take them from waterway.ditch) as they refer to the tunnel/culvert

RedAuburn replied:
oops, i think i accidentally pasted that 😅 fixed

pastk commented on data/styles/vehicle/include/priorities_2_BG-top.prio.txt:
```diff
@@ -20,8 +20,11 @@ landuse-reservoir                                   # area z12- (also has captio
 natural-water                                       # area z1- (also has caption z10-)
 natural-water-basin                                 # area z1- (also has caption z10-)
+natural-water-ditch                                 # area z1- (also has caption z10-)
+natural-water-drain
```
its good to include automatic comments changes into the main commit also (not into "regenerate");
the rule of thumb is - if there were manual changes to the priority files (in this case - new types added) then include into the main commit, if prio files changes were purely automatic (e.g. type's visibility changed in *.mapcss) then include them into the "regenerate" commit

I know its a bit of a fine line :)

RedAuburn replied:
ah ok, will keep in mind for the future :)

pastk commented:
Also - if you feel like it! - maybe experiment with the fill color of areal water=ditch/drain - its supposed to be dirty, non-drinkable water, so it would be good to visually distinguish from rivers, ponds, lakes..

vng commented on data/mapcss-mapping.csv:
```diff
@@ -513,12 +513,12 @@ healthcare|speech_therapist;438;
-deprecated|deprecated;446;x
-deprecated:railway|yard:06.2023;447;x
+natural|water|drain;[natural=water][water=drain];;name;int_name;446;
+natural|water|wastewater;[natural=water][water=wastewater];;name;int_name;447;
```
Hm, honestly saying putting a new type instead of deprecated 2 months ago is a bit scary :)
Please, keep 447 as-is, assign the new type with 448.

pastk commented on data/strings/types_strings.txt:
```diff
@@ -5014,6 +5055,47 @@
+    pt = Águas residuais
+    pt-BR = Águas residuais
+    ro = Ape uzate
+    ru = Сточные Воды
```
`ru = Сточные воды`

pastk commented on data/styles/clear/include/Basemap.mapcss:
```diff
@@ -238,6 +236,11 @@ area|z15-[natural=water][tunnel],
+area|z14-[natural=water][water=ditch],
+area|z14-[natural=water][water=drain],
+area|z14-[natural=water][water=wastewater],
+{fill-color: @water_bad;}
```
This way it'll make the color change between -z13 and z14-.
So either change it all the way or remove area fills for z1-z13 (makes sense!)

RedAuburn replied:
I've changed it so the visibility roughly matches normal water types 👍

pastk commented on data/styles/clear/include/Basemap.mapcss:
```diff
@@ -287,10 +290,9 @@ line|z15-[waterway=stream][intermittent=yes]
-{width: 1.8;opacity: 1;color: @river;dashes: 0.9,0.9;}
+line|z14-[waterway=ditch],
+line|z14-[waterway=drain],
+{width: 1.8;opacity: 1;color: @water_bad;dashes: 12,2;}
```
I see you've raised visibility significantly. Very good for outdoors, but won't it create issues in cities?
Either test extensively or raise 1 level max at a time :)

Also, could you please add a screenshot of the changed pattern and color too?

RedAuburn replied:
i don't think it'll be an issue, streams render from z14 and ditch/drain just as a line are very faint, but are still useful to see

pastk commented on data/styles/clear/include/Basemap.mapcss:
```diff
@@ -599,6 +601,10 @@ line|z19-[natural=earth_bank],
 
 
+line|z17-[barrier=ditch],
+{width: 1.8;opacity: 1;color: @cliff;dashes: 0.9,0.9;}
```
I think raising visibility of barrier=ditch is more justified, as its *designed* to be an obstacle, so got to be visible.

RedAuburn replied:
Raised it to match `man_made=embankment` (z16-)

pastk reviewed (changes requested):
Thanks for adding wastewater too! Its strange to see clean ponds by the wasterwater plants now :)

pastk commented:
Let's finish it?

RedAuburn commented:
> Let's finish it?

sorry yes, i will get to it later today hopefully :)

RedAuburn commented:
fixed the review and improved the style/colours. now it looks like this:

![image1](https://github.com/organicmaps/organicmaps/assets/26939824/ecfe0655-f016-415b-bc02-c8a01f44dad7)

pastk commented:
TBH the color difference is too little now, it should be more obvious.
It was not easy to see the difference on your screenshot even where there is a clean water river and wastewater reservoirs are side-by-side. When the wastewater thing will be alone I wouldn't be able to tell its not clean :)
